### DISCUSSION TITLE
CVRPTW Error sign when NotMemberCt is used and start/end locations are different

### DISCUSSION THREAD

**Discussion by bobome-ola** [google/or-tools#2987] [Routing (and legacy CP) questions]
Hi,

The issue concerns the implementation of NotMemberCt in conjonction with setting start/end locations (CVRPTW model).
Version: 8.1.8487
Python: 3.9.1

The current CVRPTW model is based on the [cvrptw.py example](https://github.com/google/or-tools/blob/stable/ortools/constraint_solver/samples/cvrptw.py), the [routing documentation for setting start and end locations](https://developers.google.com/optimization/routing/routing_tasks#setting-start-and-end-locations-for-routes) and the [following thread](https://github.com/google/or-tools/issues/456) regarding NotMemberCt.

After several tests, with identical time windows and applying the same NotMemberCt for all nodes:
- When start and end locations are the same, solutions are provided in respect to the multiple time windows per node.
- When start and end locations are different, there is no solution / no error message. Just an error sign.

The only way to avoid the issue is to remove the NotMemberCt applying to the end node.

Is there something I'm missing regarding that behavior?

Thanks

**Comment by Mizux** [comment]
Concerning `NotMemberCt()`, I don't know, I need to check...

But you can use `RemoveInterval()` instead 
see: https://github.com/Mizux/tsp_multiple_days/blob/feature/morning_nodes/40_Locations_3_days.py for usage

**Comment by Mizux** [comment]
Be careful, you must provide solver index not your indices so you must use `manager.NodeToIndex(your_node_index)`

Also `NodeToIndex(location)` is not defined for end node you must use `routing.End(vehicle_idx)`

**Comment by bobome-ola** [comment]
Thank you for your reply Mizux :). I will check `RemoveInterval()`.

To follow up with your comment, I'm currently using:
- `CumulVar(manager.NodeToIndex(my_node_index))` for `NotMemberCt()`
- start and end nodes are both identified when applying the window constraints for all nodes with `location_idx ==`
- `routing.End(my_vehicle_id)` is used in combination with `SetRange(data['my_time_windows'][1][0], data['my_time_windows'][1][1])`
